app/views/handlers/list_repeating_question.py

```diff
@@ -14,13 +14,37 @@ def get_next_location_url(self) -> str:
         if self._is_returning_to_section_summary():
             return self.get_section_summary_url()
 
+        if first_incomplete_location := self.get_first_incomplete_repeating_block_url(
+            self.current_location.list_item_id
+        ):
+            return first_incomplete_location
+
+        if (
+            not first_incomplete_location
+            and self.parent_block["type"] == "ListCollectorContent"
```

Comment:
Hmmm, I'm guessing the issue here is that the ListCollectorContent block is being marked as complete as soon as you go past it, and therefore unlike list collectors, you don't go back to the parent and say that you don't have more items to add?

I think this could be problematic as we're bypassing an incomplete block being marked as complete by just routing back to it each time - I reckon you probably need to do something like list collectors `_is_list_collector_complete` so that this works properly

Reply:
Addressed now, used exactly this with a small tweak (to get the list name) and it works like a charm! thanks.